src/controllers/userController.ts

```diff
@@ -2,33 +2,65 @@ import express, { Request, Response } from "express";
 import bcrypt from "bcrypt";
 import jwt from "jsonwebtoken";
 import User from "../models/User";
+import { z } from "zod";
+
+const UserSchema = z.object({
+  email: z.string().email(),
+  password: z.string().min(8),
+  mobileNumber: z
+    .string()
+    .regex(/^\d{10}$/)
+    .optional()
+    .nullable(),
+  userName: z.string().nonempty(),
+});
 
 export const register = async (req: Request, res: Response) => {
   const { email, password, mobileNumber, userName } = req.body;
-  try {
-    // const existingEmail = await User.findAll;
+  const validationResult = UserSchema.safeParse({
+    email,
+    password,
+    mobileNumber,
+    userName,
+  });
 
-    const existingUserName = await User.findAll({
+  if (!validationResult.success) {
+    // Return validation errors to the client
+    return res.status(400).json({ errors: validationResult.error.format() });
```

Comment:
Add in middleware.
And dont user hardcode (status code) 
Add in constant file. and use it from there.